tests/integration/rdb.tcl

```diff
@@ -114,13 +114,37 @@ start_server [list overrides [list "dir" $server_path] keep_persistence true] {
     r del stream
 }
 
+package require Tclx
+set old_umask [umask]
+set old_perm 666
+incr old_perm -$old_umask
+set dump_path [file join $server_path dump.rdb]
+assert_equal [file attributes $dump_path -permissions] 00$old_perm
+
+if {$old_umask == 22} {
+    set new_umask 2
+} else {
+    set new_umask 22
+}
+set new_perm 666
+incr new_perm -$new_umask
+
+umask $new_umask
+start_server [list overrides [list "dir" $server_path] keep_persistence true] {
+    test {Test nondefault umask applied} {
+        r save
+        assert_equal [file attributes $dump_path -permissions] 00$new_perm
```

Comment:
I tested this command from the Tcl shell in Tcl 9.0. It has a slightly different string output with a lowercase "o".

```
$ tclsh
% file attributes dump.rdb -permissions
0o644
```

Apparently, this is some new way to write octal numbers. It's not string-equal to 0064 but it's numerically equal. `assert_equal` uses string equality though. To accept both variants we could change to numeric comparison. Something like...

```suggestion
        assert {[file attributes $dump_path -permissions] == 00$new_perm}
```

Reply:
... or assert_range, which uses >= and <=

```suggestion
        # Use numeric comparison for compatibility with Tcl 8 and 9.
        assert_range [file attributes $dump_path -permissions] 00$new_perm 00$new_perm
```